tide/master/rest/RestServer.cpp

```diff
@@ -50,15 +53,55 @@ RestServer::RestServer(const int port)
     : zeroeq::http::Server{zeroeq::URI{QString(":%1").arg(port).toStdString()}}
 {
     _init();
+    addToWhitelist("127.0.0.1");
+}
+
+std::future<http::Response> RestServer::respondTo(
+    zeroeq::http::Request& request) const
+{
+    if (_blockedMethods.count(request.method) &&
+        !_isWhitelisted(request.source.substr(0, request.source.find(":"))))
+    {
+        return zeroeq::http::make_ready_response(http::Code::FORBIDDEN);
+    }
+    return Server::respondTo(request);
 }
 
 int RestServer::getPort() const
 {
     return getURI().getPort();
 }
 
+void RestServer::block(const zeroeq::http::Method method)
+{
+    _blockedMethods.insert(method);
+}
+
+void RestServer::unblock(const zeroeq::http::Method method)
+{
+    _blockedMethods.erase(method);
+}
+
 void RestServer::_init()
 {
     _socketNotifier.connect(&_socketNotifier, &QSocketNotifier::activated,
-                            [this]() { receive(0 /* non-blocking receive*/); });
+                            [this]() {
+                                while (receive(0 /* non-blocking receive*/))
+                                    ;
+                            });
+}
+
+void RestServer::addToWhitelist(const std::string& address)
+{
+    _whitelist.insert(address);
+}
+
+void RestServer::removeFromWhitelist(const std::string& address)
+{
+    _whitelist.erase(address);
+}
+
+bool RestServer::_isWhitelisted(const std::string& source) const
+{
+    return _whitelist.count(source.substr(0, source.find(":")));
```

Comment:
source.substr(0, source.find(":")) needs to be in a separate method _getHostname(source)